Add configuration_url to hunterdouglas_powerview

kingy444 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
Add 'Visit Device' to Hunterdouglas Powerview

## Type of change
- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

kingy444 committed:
Add Top/Down Bottom/Up support to powerview

kingy444 committed:
Add signal strength sensor to powerview

kingy444 committed:
Fix creating battery sensor on hardwired blinds and optional creation

kingy444 committed:
Rebase to dev

kingy444 committed:
Merge branch 'home-assistant-dev' into add-powerview-topdown
Alighining branch to master dev branch

kingy444 committed:
Merge pull request #1 from home-assistant/dev

Fetch changes from dev

kingy444 committed:
Rebase to dev

kingy444 committed:
Merge branch 'home-assistant-dev' into add-powerview-topdown

kingy444 committed:
Merge pull request #2 from home-assistant/dev

Merge dev

kingy444 committed:
Remove options flow and set disabled by default

kingy444 committed:
Merge branch 'add-powerview-topdown' of https://github.com/kingy444/core into add-powerview-topdown

kingy444 committed:
Resolve missed options flow code

kingy444 committed:
Update homeassistant/components/hunterdouglas_powerview/sensor.py

Co-authored-by: J. Nick Koston <nick@koston.org>

kingy444 committed:
Update homeassistant/components/hunterdouglas_powerview/sensor.py

Co-authored-by: J. Nick Koston <nick@koston.org>

kingy444 committed:
Set constant and move to attribute from property

kingy444 committed:
Fix shade matching to description

kingy444 committed:
Remove unused logging

kingy444 committed:
Move Top Down Shades to own class

kingy444 committed:
Handle empty data from 204/423 response

kingy444 committed:
Move signalStrength to constant

kingy444 committed:
Align upstream

kingy444 committed:
Merge branch 'home-assistant-dev' into add-powerview-topdown

kingy444 committed:
Extend timeout for hub maintenance

kingy444 committed:
Move top shade to own class

kingy444 committed:
Merge Upstream

kingy444 committed:
Merge branch 'home-assistant-dev' into add-powerview-topdown

kingy444 committed:
Merge conflicts to upstream

kingy444 committed:
Merge upstream changes

kingy444 committed:
Merge branch 'home-assistant-dev' into add-powerview-topdown

kingy444 committed:
Merge pull request #5 from home-assistant/dev

Merge Upstream

kingy444 committed:
Catch if all shades are hardwired

kingy444 committed:
Merge branch 'add-powerview-topdown' of https://github.com/kingy444/core into add-powerview-topdown

kingy444 committed:
Fix close cover for top shade

kingy444 committed:
Remove changes unrelated to tdbu

kingy444 committed:
Add BatterKind back

kingy444 committed:
Apply suggestions per pull request

bdraco committed:
Merge branch 'dev' into add-powerview-topdown

kingy444 committed:
Move to coordinator values

kingy444 committed:
Merge branch 'add-powerview-topdown' of https://github.com/kingy444/core into add-powerview-topdown

kingy444 committed:
Fix current vane position

kingy444 committed:
Better handling of tdbu shades

bdraco committed:
Merge branch 'dev' into add-powerview-topdown

bdraco committed:
Merge branch 'dev' into add-powerview-topdown

kingy444 committed:
Update Clamping

kingy444 committed:
Add visit device

kingy444 committed:
Merge branch 'home-assistant:dev' into add-powerview-diagnostics

kingy444 committed:
Move to constant

kingy444 committed:
Merge branch 'add-powerview-diagnostics' of https://github.com/kingy444/core into add-powerview-diagnostics

kingy444 committed:
Add Diagnostics

kingy444 committed:
Merge upstream changes

kingy444 committed:
Update Diagnostics to work with PowerviewShadeDataModel

kingy444 committed:
Merge branch 'home-assistant-dev' into add-powerview-diagnostics

kingy444 committed:
Merge pull request #6 from home-assistant/dev

Merge upstream

epenet commented:
A few notes:
- could you split this PR into 2 PRs (one for diagnostics and one for visit)? smaller PR means easier review and less time in the queue
- for diagnostics, can you either add some tests, or add to `.coveragerc` and show a sample file?
- in the PR description, ensure that you place the x inside the `[ ]` without spaces to show correctly as tick-box

kingy444 committed:
Remove diagnostics

kingy444 commented:
> A few notes:
> 
> * could you split this PR into 2 PRs (one for diagnostics and one for visit)? smaller PR means easier review and less time in the queue
> * for diagnostics, can you either add some tests, or add to `.coveragerc` and show a sample file?
> * in the PR description, ensure that you place the x inside the `[ ]` without spaces to show correctly as tick-box

Removed diagnostics and reverted this to just visit device

epenet commented on homeassistant/components/hunterdouglas_powerview/entity.py:
```diff
@@ -62,6 +69,7 @@ def device_info(self) -> DeviceInfo:
             name=self._device_info[DEVICE_NAME],
             suggested_area=self._room_name,
             sw_version=sw_version,
+            configuration_url=f"http://{self.hub_address}/api/shades",
```
```suggestion
            configuration_url=f"http://{self._hub_address}/api/shades",
```

epenet commented on homeassistant/components/hunterdouglas_powerview/entity.py:
```diff
@@ -97,6 +105,7 @@ def device_info(self) -> DeviceInfo:
             manufacturer=MANUFACTURER,
             model=str(self._shade.raw_data[ATTR_TYPE]),
             via_device=(DOMAIN, self._device_info[DEVICE_SERIAL_NUMBER]),
+            configuration_url=f"http://{self.hub_address}/api/shades/{self._shade.id}",
```
```suggestion
            configuration_url=f"http://{self._hub_address}/api/shades/{self._shade.id}",
```

kingy444 committed:
Remove hub_address property

kingy444 commented on homeassistant/components/hunterdouglas_powerview/entity.py:
```diff
@@ -40,8 +41,14 @@ def __init__(
+        self._hub_address = device_info[PV_HUB_ADDRESS]
         self._device_info = device_info
 
+    @property
+    def hub_address(self):
+        """Return the ip of the hub."""
+        return self._hub_address
+
```
Felt like there was a reson i did it this way but tested and its working as suggested. Change pushed

epenet reviewed (approved):
Code LGTM (but I don't have powerview to test)

bdraco commented on homeassistant/components/hunterdouglas_powerview/__init__.py:
```diff
@@ -72,6 +73,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     try:
         async with async_timeout.timeout(10):
             device_info = await async_get_device_info(pv_request)
+            device_info[PV_HUB_ADDRESS] = hub_address
```
We should make `device_info` a dataclass in the future

bdraco commented on homeassistant/components/hunterdouglas_powerview/__init__.py:
```diff
@@ -97,6 +99,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     coordinator = PowerviewShadeUpdateCoordinator(hass, shades, hub_address)
     coordinator.async_set_updated_data(PowerviewShadeData())
+
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
```
In a future PR we should create a dataclass in a `models.py` instead of using a `dict` here

bdraco committed:
Merge branch 'dev' into add-powerview-diagnostics
